src/service/performance-impl.js

```diff
@@ -233,6 +269,48 @@ export class Performance {
     observer.observe({entryTypes: entryTypesToObserve});
   }
 
+  /**
+   * Bind to various page lifecycle events to capture visibility state changes.
+   * @see https://developers.google.com/web/updates/2018/07/page-lifecycle-api
+   */
+  registerLayoutJankHandler_() {
+    // The visibility state of the document.
+    let visibilityState = getVisibilityState(this.win.document);
+    // Whether the `lj` metric was sent.
+    let sentFirstJankScore = false;
+    // Whether the `lj-2` metric was sent.
+    let sentSecondJankScore = false;
+
+    const handleVisibilityStateChange = nextState => {
+      const previousState = visibilityState;
+      if (previousState === nextState) {
+        return;
+      }
+      visibilityState = nextState;
+
+      if (nextState !== 'hidden') {
+        return;
+      }
+
+      if (!sentFirstJankScore) {
+        this.tickDelta('lj', this.aggregateJankScore_);
+        this.flush();
+        sentFirstJankScore = true;
+        return;
+      }
+      if (!sentSecondJankScore) {
+        this.tickDelta('lj-2', this.aggregateJankScore_);
+        this.flush();
+        sentSecondJankScore = true;
+        return;
```

Comment:
At this point should you remove the event listeners?

Reply:
Done, partially. Adding a comment on new code about issue in tests.